common/src/test/scala/uk/ac/wellcome/sqs/SQSReaderTest.scala

```diff
@@ -0,0 +1,53 @@
+package uk.ac.wellcome.sqs
+
+import com.amazonaws.services.sqs.model.{Message, ReceiveMessageResult}
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatest.{FunSpec, Matchers}
+import uk.ac.wellcome.models.aws.SQSConfig
+import uk.ac.wellcome.test.utils.SQSLocal
+
+import scala.collection.JavaConversions._
+import scala.concurrent.duration._
+
+class SQSReaderTest
+    extends FunSpec
+    with Matchers
+    with ScalaFutures
+    with IntegrationPatience
+    with SQSLocal {
+
+  it("should get messages from the SQS queue, limited by the maximum number of messages and return them") {
+    val sqsConfig =
+      SQSConfig("eu-west-1", idMinterQueueUrl, waitTime = 20 seconds, 2)
+    val messageStrings = List("someMessage1", "someMessage2", "someMessage3")
+    messageStrings.foreach(sqsClient.sendMessage(idMinterQueueUrl, _))
+    val sqsReader =
+      new SQSReader(sqsClient, sqsConfig)
+
+    val futureMessages = sqsReader.retrieveMessages()
+
+    whenReady(futureMessages) { messages =>
+      messages should have size 2
```

Comment:
Maybe worth a quick comment here about why we aren’t doing the more obvious test?

> SQS isn’t a FIFO queue, so we can’t just check we get "someMessage1" and "someMessage2". Instead we have to test that we receive two distinct messages, both of which were in the original set of messages.